### PR TITLE
upgrade(cli): MUD Version 2.0.0-next.8

### PR DESCRIPTION
**Pull request by aLIEzsss4**

### CODE REVIEW

**Comment by aLIEzsss4** [comment]
The contracts have been split for now, further optimization is needed.

File: package.json [hunk 1]
```diff
@@ -7,8 +7,8 @@
     "dev:client": "pnpm --filter 'client' run dev",
     "dev:contracts": "pnpm --filter 'contracts' dev",
     "foundry:up": "curl -L https://foundry.paradigm.xyz | bash && bash $HOME/.foundry/bin/foundryup",
-    "mud:up": "pnpm recursive exec mud set-version -v canary && pnpm install",
-    "prepare": "(forge --version || pnpm foundry:up);pnpm simple-git-hooks",
```
Comment (noyyyy) [inline package.json:11]: why remove `simple-git-hooks`. It's used for code auto format

File: packages/contracts/remappings.txt [hunk 2]
```diff
@@ -1,4 +1,5 @@
-forge-std/=lib/forge-std/src/
-@latticexyz/=node_modules/@latticexyz/
+ds-test/=./node_modules/ds-test/src/
+forge-std/=./node_modules/forge-std/src/
```
Comment (noyyyy) [inline packages/contracts/remappings.txt]: don't use `forge-std`, `ds-test`. Use forge install to install them

File: packages/indexer/Dockerfile [hunk 3]
```diff
@@ -11,7 +11,7 @@ WORKDIR /app
 ## clone mud repo and lock hash
 RUN git clone https://github.com/latticexyz/mud.git && \
     cd mud && \
-    git checkout 57ee7f9cb7b5803a6f6b6ad69a7d50fbef4d8dc1 && \
+    git checkout 1efccab99fbb5b16a5e7980683f74f3ad8ec4e21 && \
```
Comment (noyyyy) [inline packages/indexer/Dockerfile]: Does indexer code change? Do you deploy and test it?

**Review by noyyyy — APPROVED** [review]

**Review by noyyyy — CHANGES_REQUESTED** [review]
see above

**Comment by noyyyy** [comment]
image
https://discord.com/channels/865335009915961364/1151938472131694745/1152120768310820895

**Comment by aLIEzsss4** [comment]
This MUD upgrade includes development versions of forge-std and ds-test dependencies. I think we should synchronize with them. 

https://github.com/latticexyz/mud/blob/5150544354816949ab3d47950df7fc67ded797a7/examples/minimal/packages/contracts/package.json#L27

or 
```
forge-std/=lib/forge-std/src/
ds-test/=lib/ds-test/src/
@latticexyz/=node_modules/@latticexyz/
@openzeppelin-contracts=lib/openzeppelin-contracts/contracts
cement/=lib/cement/contracts/
```

**Comment by noyyyy** [comment]
> This MUD upgrade includes development versions of forge-std and ds-test dependencies. I think we should synchronize with them.
> 
> https://github.com/latticexyz/mud/blob/5150544354816949ab3d47950df7fc67ded797a7/examples/minimal/packages/contracts/package.json#L27
> 
> or
> 
> ```
> forge-std/=lib/forge-std/src/
> ds-test/=lib/ds-test/src/
> @latticexyz/=node_modules/@latticexyz/
> @openzeppelin-contracts=lib/openzeppelin-contracts/contracts
> cement/=lib/cement/contracts/
> ```

If you use ssh clone. It will generate lock which cannot be used directly in github action. 

If you have used foundry, foundry project mainly use forge install to generate git submodule. By using submodule, you can also assign a commit hash.

Also the `forge-std` and `ds-test` are not maintained by lattice. Their api is stable. It usually doesn't matter if you use a different newer version.

**Comment by aLIEzsss4** [comment]
> > This MUD upgrade includes development versions of forge-std and ds-test dependencies. I think we should synchronize with them.
> > https://github.com/latticexyz/mud/blob/5150544354816949ab3d47950df7fc67ded797a7/examples/minimal/packages/contracts/package.json#L27
> > or
> > ```
> > forge-std/=lib/forge-std/src/
> > ds-test/=lib/ds-test/src/
> > @latticexyz/=node_modules/@latticexyz/
> > @openzeppelin-contracts=lib/openzeppelin-contracts/contracts
> > cement/=lib/cement/contracts/
> > ```
> 
> If you use ssh clone. It will generate lock which cannot be used directly in github action.
> 
> If you have used foundry, foundry project mainly use forge install to generate git submodule. By using submodule, you can also assign a commit hash.
> 
> Also the `forge-std` and `ds-test` are not maintained by lattice. Their api is stable. It usually doesn't matter if you use a different newer version.

fix

File: packages/indexer/fly.alt.toml [hunk 4]
```diff
@@ -27,4 +27,4 @@ RPC_HTTP_URL = "https://flashlayer.alt.technology/autochessia37806fd60"
 RPC_WS_URL = "wss://flashlayer.alt.technology/autochessia37806fd60"
 SQLITE_FILENAME = "/data/alt.db"
 CHAIN_ID = 1398383
-START_BLOCK = 702262
```
Comment (noyyyy) [inline packages/indexer/fly.alt.toml]: just discard this change.

File: packages/indexer/fly.lattice.toml [hunk 5]
```diff
@@ -24,4 +24,4 @@ destination = "/data"
 [env]
 SQLITE_FILENAME = "/data/lattice.db"
 CHAIN_ID = 4242
-START_BLOCK = 23288565
```
Comment (noyyyy) [inline packages/indexer/fly.lattice.toml]: just discard this change.

**Review by noyyyy — CHANGES_REQUESTED** [review]
I will modify indexer things in another pr. Discard these to avoid conflict.

**Comment by aLIEzsss4** [comment]
> I will modify indexer things in another pr. Discard these to avoid conflict.

fix

**Review by noyyyy — APPROVED** [review]
LGTM

**Comment by noyyyy** [comment]
Let's work together when you're free @aLIEzsss4. We need another indexer change pr to be merged at the same time

**Comment by noyyyy** [comment]
See #184, but it's wip.

**Pull request merged**